instrumentation/opentelemetry-instrumentation-botocore/tests/test_botocore_instrumentation.py

```diff
@@ -629,3 +629,157 @@ def test_dynamodb_client(self):
+            expected_result_keys,
+            get_item_attributes.get(response_hook_result_attribute_name),
+        )
+
+    @mock_dynamodb2
+    def test_hook_name_format(self):
+        def request_hook(span, operation_name, api_params):
+            if span and span.is_recording():
+                span.set_attribute("request_hook_was_called", True)
+
+        def response_hook(span, result):
+            if span and span.is_recording():
+                span.set_attribute("response_hook_was_called", True)
+
+        BotocoreInstrumentor().uninstrument()
+
+        # Hook names format is either {service_name}_request_hook or {service_name}_response_hook
+        # where service_name should be lowercase with no dashes
+        # for example: dynamodb_request_hook, sns_response_hook, etc...
+
+        # The instrumentor will ignore these hook since they have bad name format
+        BotocoreInstrumentor().instrument(
+            dynamo_db_request_hook=request_hook,
+            response_hook_dynamodb=response_hook,
+        )
+
+        self.session = botocore.session.get_session()
+        self.session.set_credentials(
+            access_key="access-key", secret_key="secret-key"
+        )
+
+        ddb = self.session.create_client("dynamodb", region_name="us-west-2")
+
+        ddb.create_table(
+            AttributeDefinitions=[
+                {"AttributeName": "id", "AttributeType": "S"},
+            ],
+            KeySchema=[{"AttributeName": "id", "KeyType": "HASH"}],
+            ProvisionedThroughput={
+                "ReadCapacityUnits": 5,
+                "WriteCapacityUnits": 5,
+            },
+            TableName="test_table",
+        )
+
+        spans = self.memory_exporter.get_finished_spans()
+        assert spans
+        self.assertEqual(len(spans), 1)
+        span_attributes = spans[0].attributes
+
+        self.assertFalse(span_attributes.get("request_hook_was_called", False))
+        self.assertFalse(
+            span_attributes.get("response_hook_was_called", False)
+        )
+
+        # Now with the good naming format, hooks should be called
+        BotocoreInstrumentor().uninstrument()
+        BotocoreInstrumentor().instrument(
+            dynamodb_request_hook=request_hook,
+            dynamodb_response_hook=response_hook,
+        )
+
+        ddb.create_table(
+            AttributeDefinitions=[
+                {"AttributeName": "id", "AttributeType": "S"},
+            ],
+            KeySchema=[{"AttributeName": "id", "KeyType": "HASH"}],
+            ProvisionedThroughput={
+                "ReadCapacityUnits": 5,
+                "WriteCapacityUnits": 5,
+            },
+            TableName="test_table_2",
+        )
+
+        spans = self.memory_exporter.get_finished_spans()
+        assert spans
+        self.assertEqual(len(spans), 2)
+        span_attributes = spans[1].attributes
+
+        self.assertTrue(span_attributes.get("request_hook_was_called", False))
```

Comment:
shouldn't this be True?

Reply:
it should, I use `assertTrue` here
the `False` here is the second parameter for the `get` method (to be returned if the key is missing, instead of `None`), not for the assert function

but anyway, I removed this test, since I don't expect a specific name format anymore